src/python_testing/TC_DISHALM_3_1.py

```diff
@@ -0,0 +1,143 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class TC_DISHALM_3_1(MatterBaseTest):
+
+    async def read_da_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.DishwasherAlarm
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def subscribe_event(self, endpoint):
+        return await self.default_controller.SubscribeEvent(self.dut_node_id, [(endpoint, Clusters.DishwasherAlarm.Events.Notify)])
+
+    async def read_event_notify_state(self, endpoint):
+        return await self.default_controller.ReadEvent(self.dut_node_id, [(endpoint, Clusters.DishwasherAlarm.Events.Notify.state)])
+
+    async def read_event_notify(self, endpoint):
+        return await self.default_controller.ReadEvent(self.dut_node_id, [(endpoint, Clusters.DishwasherAlarm.Events.Notify)])
+
+    async def read_state_attribute(self, endpoint):
+        return await self.read_da_attribute_expect_success(endpoint, Clusters.DishwasherAlarm.Attributes.State)
+
+    async def read_latch_attribute(self, endpoint):
+        return await self.read_da_attribute_expect_success(endpoint, Clusters.DishwasherAlarm.Attributes.Latch)
+
+    async def send_reset_cmd(self, setAlarm) -> Clusters.Objects.DishwasherAlarm.Commands.Reset:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.DishwasherAlarm.Commands.Reset.alarms(setAlarm=setAlarm), endpoint=self.endpoint)
+        return ret
+
+    @async_test_body
+    async def test_TC_DISHALM_3_1(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        asserts.assert_true(self.check_pics("DISHALM.S.A0001"), "DISHALM.S.A0001 must be supported")
+
+        self.print_step("1a", "Commissioning, already done")
+
+        self.print_step("1b", "Set up subscription to Notify Event")
+        await self.subscribe_event(endpoint=endpoint)
+
+        if self.check_pics("DISHALM.S.A0002"):
+            self.print_step("1c", "TH reads from the DUT the Latch attribute")
+            latch_response = self.read_latch_attribute(endpoint=endpoint)
+            logging.info("Latch: %s" % (latch_response))
+        else:
+            self.print_step("1d", "Optional Latch attribute not supported, set LatchResponse to 0")
+            latch_response = 0
+
+        self.print_step("2a", "Operate Device to set the condition to raise the Inflow alarm and wait a few seconds")
+        input("Press Enter when done.\n")
+
+        notify = self.read_event_notify(endpoint=endpoint)
+        notify_state = self.read_event_notify_state(endpoint=endpoint)
```

Comment:
Test plan has subscription reports here, not event reads, no?  Or do I misunderstand what `read_event_notify` and `read_event_notify_state` do?

This was a question last review too...